### DISCUSSION TITLE
GH Label Cleanup

### DISCUSSION THREAD

**Discussion by alexanderbez** [osmosis-labs/osmosis#1134] [Ideas]
I'd like to propose a bit of housekeeping and general cleanup to our Github label management. This is based on my experience that I've observed working with Tendermint, the Cosmos SDK and applications built on top.

To summarize

- Remove legacy backport labels and mergify configurations for versions that we obviously no longer support (i.e. <= v7)
- Remove any unused  and miscellaneous labels
- For the labels that remain, introduce two primary prefixes to most, if not all, labels:
  - `C:` which stands for component/category
    - e.g. `C:x/gamm`, `C:x/epoch`, `C:IBC`
  - `T:` which stands for type
    - e.g. `T:Bug`, `T:CI`, `T:Security`, `T:Docs`
  - `S:` which stands for status
    - e.g. `S:backport/v7.x.x`, `S:Blocked`, `S:Pinned`
  - `P:` which stands for priority
    - e.g. `P:High`, `P:Medium`, `P:Low`
  - `A:` which stands for action
    - e.g. `A:Automerge`
  - All label types are color-coded, e.g. blue for `C` and purple for `T`
  - Override colors for context-dependent labels, e.g. red for `P:Critical` or `T:Bug`

And that's pretty much it. Ideally, we have no more than 30-40 labels if we do this correctly, maybe even less.

Examples:
- https://github.com/umee-network/umee/labels
- https://github.com/tendermint/tendermint/labels

**Comment by daniel-farina** [comment]
Thank you @alexanderbez  for the label improvement ideas!

I agree with the label structure for things like:
  - `C:` which stands for component/category
    - e.g. `C:x/gamm`, `C:x/epoch`, `C:IBC`
  - `T:` which stands for type
    - e.g. `T:Bug`, `T:CI`, `T:Security`, `T:Docs`

I don't think the `S:`, `P`, etc are not needed as we are managing these items at a project level via custom fields.  The labels are used due to a lack of functionality in Github's UI but now there is better tooling such Github Projects Beta which can help make this process a bit cleaner. 

<img width="1497" alt="image" src="https://user-images.githubusercontent.com/13665117/159781781-c3c029cd-41be-4e2c-93c4-c7459c12e998.png">

I think we need to keep a good balance between labels and project's custom fields. I don't think we want too many labels or too many custom fields.  Things like   `C:x/gamm`, `C:x/epoch`, `C:IBC`  will definitely make the labeling experience much better. I would like to follow up on this on tomorrow's call. 

Thank you!

**Reply by alexanderbez** [reply]
Sure, that I think makes sense. But I still think there's value in those labels in certain context, especially the label `S:backport/vN.x.x`. If you want, I suppose you can apply a `T:` to this, but I really do think there's merit to `S:` prefixes, albeit we can drop the status ones. 

e.g. I'd like to view an issue in isolation w/o having to go to the project view to understand various context about it. WDYT?

**Reply by alexanderbez** [reply]
Also, you need `A:` too for automated github and workflow actions IMO. E.g. it should be `A:Automerge`, not `automerge`.

**Comment by p0mvn** [comment]
Can we remove the icons from labels or move them to the end of the label name? Makes it less trivial searching by labels when they are in the front

**Reply by p0mvn** [reply]
For example:
![image](https://user-images.githubusercontent.com/34196718/159819451-0d2a5f82-80f5-4901-815a-93816b2fdc0f.png)

There are more "frogs" with an icon in the front that are not displayed.

**Reply by alexanderbez** [reply]
Yeah, good idea!

**Comment by ValarDragon** [comment]
Per chain dev call, going with `C`, `T` and `A` labels. Also moving the emojis to the end of labels for better search

Thanks for this suggestion Bez!
